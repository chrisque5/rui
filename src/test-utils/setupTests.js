// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
