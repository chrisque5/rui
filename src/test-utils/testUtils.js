/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
