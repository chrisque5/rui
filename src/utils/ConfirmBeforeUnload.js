import { useEffect, useCallback } from 'react';

const ConfirmBeforeUnload = () => {
  const onBeforeUnload = useCallback((e) => {
    e.preventDefault();
    e.returnValue = '';
  }, []);

  useEffect(() => {
    window.addEventListener('beforeunload', onBeforeUnload);

    return () => {
      window.removeEventListener('beforeunload', onBeforeUnload);
    };
  }, [onBeforeUnload]);

  return null;
};

export default ConfirmBeforeUnload;
