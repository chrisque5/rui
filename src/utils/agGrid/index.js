import {
  approvalsCellRenderer, approvalsSummaryCellRenderer, isUnderInvestigationCellRenderer, stpCellRenderer, stpFilterCellRenderer,
} from './js-renderers';
import {
  dateFormatter, formatDate, numberFormatter, timeFormatter,
  valueDateFormatter, fixingDateFormatter, publishDateFormatter,
  priceFormatter, brokerageFormatter,
} from './formatters';
import { dateFilterComparator, numberFilterComparitor, timeFilterComparator } from './comparitors';
import { getExportMenuItems } from './exportUtils';

export {
  approvalsCellRenderer,
  approvalsSummaryCellRenderer,
  dateFilterComparator,
  dateFormatter,
  formatDate,
  getExportMenuItems,
  isUnderInvestigationCellRenderer,
  numberFilterComparitor,
  numberFormatter,
  stpCellRenderer,
  stpFilterCellRenderer,
  timeFilterComparator,
  timeFormatter,
  valueDateFormatter,
  fixingDateFormatter,
  publishDateFormatter,
  priceFormatter,
  brokerageFormatter,
};
