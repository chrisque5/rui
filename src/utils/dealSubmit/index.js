import mapFormToFwdChain from './mapFormToFwdChain';
import mapFormToNdfChain from './mapFormToNdfChain';
import mapFormToSptChain from './mapFormToSptChain';

export { mapFormToFwdChain, mapFormToNdfChain, mapFormToSptChain };
