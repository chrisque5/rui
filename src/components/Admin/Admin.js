import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Prompt, useHistory } from 'react-router-dom';
import { Tabs } from 'antd';
import { initUserData } from '../../actions/userActions';
import { resetAdminTabChanges, changePage } from '../../actions/uiActions';
import { getUser, getIsBrowserForceRefreshEnabled } from '../../utils/selectors';
import ConfirmBeforeUnload from '../../utils/ConfirmBeforeUnload';
import { ConfirmUnsavedChanges } from '../../utils/helper';
import { navMenuItems } from '../../utils/constants';

import NavbarContainer from '../Navbar/NavbarContainer';
import Forbidden from '../Error/Forbidden';
import BrokerOverrideContainer from '../BrokerOverride/BrokerOverrideContainer';
import CurrencyOverrideContainer from '../CurrencyOverride/CurrencyOverrideContainer';

const { TabPane } = Tabs;

const Admin = () => {
  const adminTabs = {
    BROKERS: 'Brokers',
    CURRENCIES: 'Currencies',
  };

  const dispatch = useDispatch();
  const history = useHistory();
  const [activeTab, setActiveTab] = useState(adminTabs.BROKERS);
  const [isTabDirty, setIsTabDirty] = useState(false);
  const { fullName, permissions = {} } = useSelector(getUser);
  const { validForDeskEdit, validForFwd } = permissions;
  const isBrowserForceRefreshEnabled = useSelector(getIsBrowserForceRefreshEnabled);

  useEffect(() => {
    dispatch(initUserData());
    dispatch(changePage(navMenuItems.ADMIN.key));

    return () => {
      dispatch(changePage(''));
    };
  }, [dispatch]);

  const checkDirtyTab = (nextTab) => {
    if (isTabDirty) {
      ConfirmUnsavedChanges(
        'You have unsaved changes in this tab. Are you sure you want to switch?',
        () => {
          dispatch(resetAdminTabChanges(true));
          setActiveTab(nextTab);
        },
      );
      return false;
    }
    setActiveTab(nextTab);
    return true;
  };

  const confirmNavigation = (location) => {
    ConfirmUnsavedChanges(
      'You have unsaved changes. Are you sure you want to leave the Admin screen?',
      () => {
        setIsTabDirty(false);
        history.push(location.pathname);
      },
    );
    return false;
  };

  return (
    <>
      {!isBrowserForceRefreshEnabled && isTabDirty && <ConfirmBeforeUnload />}
      <Prompt when={isTabDirty} message={confirmNavigation} />
      <NavbarContainer selectedKey="ADMIN" />
      {validForDeskEdit ? (
        <div className="dms-web-admin">
          <Tabs activeKey={activeTab} type="card" onChange={(nextTab) => checkDirtyTab(nextTab)}>
            <TabPane test-id={adminTabs.BROKERS} tab="Brokers" key={adminTabs.BROKERS}>
              <BrokerOverrideContainer setIsTabDirty={setIsTabDirty} />
            </TabPane>
            {validForFwd && (
              <TabPane test-id={adminTabs.CURRENCIES} tab="Currencies" key={adminTabs.CURRENCIES}>
                <CurrencyOverrideContainer setIsTabDirty={setIsTabDirty} />
              </TabPane>
            )}
          </Tabs>
        </div>
      ) : fullName && <Forbidden />}
    </>
  );
};

export default Admin;
