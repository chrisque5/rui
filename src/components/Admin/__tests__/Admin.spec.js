import { HashRouter, Route } from 'react-router-dom';
import { fireEvent, renderWithProvider, waitFor } from 'test-utils/testUtils';
import Admin from '../Admin';
import {} from '../../../utils/constants';

const initialState = {};

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    goBack: jest.fn(),
  }),
}));

jest.mock('../../../utils/selectors.js', () => ({
  getUser: jest.fn().mockReturnValue({
    permissions: {
      validForDeskEdit: true,
      validForFwd: true,
    },
    fullName: 'Test User',
    preferences: {
      settings: {},
      defaults: { defaultEntryPage: 'NA' },
    },
  }),
  getIsBrowserForceRefreshEnabled: jest.fn().mockReturnValue(false),
  getUserPreferenceSettings: jest.fn().mockReturnValue({
    colourScheme: 'light',
    dealSubmitSound: 'ka-ching',
    displayClientFavourites: true,
    displayClients: true,
    displayCurrencyFavourites: true,
    displayExecutionVenueColours: true,
    displayExecutionVenueFavourites: true,
    displayTermFavourites: true,
    lrMode: false,
    ratesFeed: true,
  }),
  getIsSettingsModalVisible: jest.fn().mockReturnValue(true),
  getUserPermissions: jest.fn().mockReturnValue({
    validForDeskEdit: true,
    validForFwd: true,
  }),
  getUserPreferencesDefaults: jest.fn().mockReturnValue({
    lastDealType: 'FWD',
    dealDefaults: [],
    defaultEntryPage: 'NDF',
  }),
  getDesks: jest.fn().mockReturnValue([
    { key: 8159, displayValue: 'BELFAST RATES TPSIN DESK' },
    { key: 8160, displayValue: 'BELFAST RATES TPUS DESK' },
  ]),
  getBrokers: jest.fn().mockReturnValue([
    { id: 100120, name: 'BELFAST TPSIN TEST BROKER 3' },
    { id: 100121, name: 'BELFAST TPSIN TEST BROKER 4' }]),
  getGcdBrokers: jest.fn().mockReturnValue([
    { id: 100118, name: 'BELFAST TPSIN TEST BROKER 1' },
    { id: 100119, name: 'BELFAST TPSIN TEST BROKER 2' },
    { id: 100120, name: 'BELFAST TPSIN TEST BROKER 3' },
    { id: 100121, name: 'BELFAST TPSIN TEST BROKER 4' },
    { id: 100122, name: 'BELFAST TPSIN TEST BROKER 5' }]),
  getBrokerUpdateStatus: jest.fn().mockReturnValue(true),
  getResetAdminTabChanges: jest.fn().mockReturnValue(false),
  getCurrencies: jest.fn().mockReturnValue([{
    baseCurrency: 'GBP',
    counterCurrency: 'USD',
    instrumentId: 4315,
    baseCurrencyDayCountYear: '365',
    counterCurrencyDayCountYear: '365',
    scalingFactor: 0.00010000,
    primaryExerciseCurrencies: '',
    secondaryExerciseCurrencies: '',
    primaryDeliveryCurrencies: 'GBP',
    secondaryDeliveryCurrencies: 'USD',
    isCLS: false,
  }, {
    baseCurrency: 'USD',
    counterCurrency: 'CNH',
    instrumentId: 4310,
    baseCurrencyDayCountYear: '360',
    counterCurrencyDayCountYear: '360',
    scalingFactor: 0.00010000,
    primaryExerciseCurrencies: 'USD',
    secondaryExerciseCurrencies: 'CNH',
    primaryDeliveryCurrencies: 'USD',
    secondaryDeliveryCurrencies: 'CNH',
    isCLS: true,
  }]),
}));

describe('<Admin />', () => {
  test('renders snapshot as expected', () => {
    const defaultProps = { isTabDirty: true };
    const { container } = renderWithProvider(
      <HashRouter>
        <Route>
          <Admin {...defaultProps} />
        </Route>
      </HashRouter>,
      { initialState },
    );
    expect(container).toMatchSnapshot();
  });

  test('switch tabs', async () => {
    const defaultProps = { isTabDirty: true };
    const { getByText } = renderWithProvider(
      <HashRouter>
        <Route>
          <Admin {...defaultProps} />
        </Route>
      </HashRouter>,
      { initialState },
    );
    const currenciesTab = await waitFor(() => getByText('Brokers'));
    fireEvent.click(currenciesTab);
  });
});
