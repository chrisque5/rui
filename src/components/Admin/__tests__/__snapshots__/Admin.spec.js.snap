// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> renders snapshot as expected 1`] = `
<div>
  <div
    class="dms-trade-capture-navbar ant-menu-dark"
  >
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark navbar-menu"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-no-link"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <img
            alt="logo"
            height="30"
            src="tpicap.png"
            style="margin-right: 15px;"
            width="30"
          />
          <span>
            Deal Management System
          </span>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal user-name"
        role="none"
        style="opacity: 1; order: 1;"
      >
        <div
          aria-controls="rc-menu-uuid-test-submenu-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <span
              class="submenu-title-wrapper"
              data-testid="ddlUserDropdown"
              id="ddlUserDropdown"
            >
              
              <span
                aria-label="caret-down"
                class="anticon anticon-caret-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="caret-down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="0 0 1024 1024"
                  width="1em"
                >
                  <path
                    d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>
  <div
    class="dms-web-admin"
  >
    <div
      class="ant-tabs ant-tabs-top ant-tabs-card"
    >
      <div
        class="ant-tabs-nav"
        role="tablist"
      >
        <div
          class="ant-tabs-nav-wrap"
        >
          <div
            class="ant-tabs-nav-list"
            style="transform: translate(0px, 0px);"
          >
            <div
              class="ant-tabs-tab ant-tabs-tab-active"
            >
              <div
                aria-controls="rc-tabs-test-panel-Brokers"
                aria-selected="true"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-Brokers"
                role="tab"
                tabindex="0"
              >
                Brokers
              </div>
            </div>
            <div
              class="ant-tabs-tab"
            >
              <div
                aria-controls="rc-tabs-test-panel-Currencies"
                aria-selected="false"
                class="ant-tabs-tab-btn"
                id="rc-tabs-test-tab-Currencies"
                role="tab"
                tabindex="0"
              >
                Currencies
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
        <div
          class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
        >
          <button
            aria-controls="rc-tabs-test-more-popup"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-hidden="true"
            class="ant-tabs-nav-more"
            id="rc-tabs-test-more"
            style="visibility: hidden; order: 1;"
            tabindex="-1"
            type="button"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-tabs-content-holder"
      >
        <div
          class="ant-tabs-content ant-tabs-content-top"
        >
          <div
            aria-hidden="false"
            aria-labelledby="rc-tabs-test-tab-Brokers"
            class="ant-tabs-tabpane ant-tabs-tabpane-active"
            id="rc-tabs-test-panel-Brokers"
            role="tabpanel"
            tabindex="0"
          >
            <form
              class="ant-form ant-form-horizontal ant-form-hide-required-mark"
            >
              <div
                class="ant-row"
                style="margin-top: 10px;"
                type="flex"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <div
                          class="ant-radio-group ant-radio-group-outline"
                          id="dealType"
                        >
                          <span
                            class="custom-label"
                          >
                            Deal Type: 
                          </span>
                          <label
                            class="ant-radio-wrapper ant-radio-wrapper-checked"
                          >
                            <span
                              class="ant-radio ant-radio-checked"
                            >
                              <input
                                checked=""
                                class="ant-radio-input"
                                data-testid="rdodealTypeNDF"
                                type="radio"
                                value="NDF"
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                            <span>
                              NDF
                            </span>
                          </label>
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                class="ant-radio-input"
                                data-testid="rdodealTypeFWD"
                                type="radio"
                                value="FWD"
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                            <span>
                              FWD
                            </span>
                          </label>
                          <label
                            class="ant-radio-wrapper"
                          >
                            <span
                              class="ant-radio"
                            >
                              <input
                                class="ant-radio-input"
                                data-testid="rdodealTypeSPT"
                                type="radio"
                                value="SPT"
                              />
                              <span
                                class="ant-radio-inner"
                              />
                            </span>
                            <span>
                              SPT
                            </span>
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row"
                style="margin-top: 10px;"
                type="flex"
              >
                <div
                  style="width: 250px;"
                >
                  <div>
                    Desks: 
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <div
                            class="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                            data-testid="ddDesk"
                            name="desk"
                          >
                            <div
                              class="ant-select-selector"
                            >
                              <span
                                class="ant-select-selection-search"
                              >
                                <input
                                  aria-activedescendant="desk_list_0"
                                  aria-autocomplete="list"
                                  aria-controls="desk_list"
                                  aria-haspopup="listbox"
                                  aria-owns="desk_list"
                                  autocomplete="off"
                                  class="ant-select-selection-search-input"
                                  id="desk"
                                  role="combobox"
                                  type="search"
                                  value=""
                                />
                              </span>
                              <span
                                class="ant-select-selection-placeholder"
                              />
                            </div>
                            <span
                              aria-hidden="true"
                              class="ant-select-arrow"
                              style="user-select: none;"
                              unselectable="on"
                            >
                              <span
                                aria-label="down"
                                class="anticon anticon-down ant-select-suffix"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="down"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                  />
                                </svg>
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                style="margin-top: 10px; margin-bottom: 5px;"
              >
                Select Brokers: 
              </div>
              <div
                class="ant-row ant-form-item"
              >
                <div
                  class="ant-col ant-form-item-control"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <div
                        class="ant-transfer"
                      >
                        <div
                          class="ant-transfer-list"
                          style="width: 380px;"
                        >
                          <div
                            class="ant-transfer-list-header"
                          >
                            <label
                              class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                            >
                              <span
                                class="ant-checkbox"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-dropdown-trigger ant-transfer-list-header-dropdown"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                            <span
                              class="ant-transfer-list-header-selected"
                            >
                              3
                               
                              items
                            </span>
                            <span
                              class="ant-transfer-list-header-title"
                            >
                              All Brokers
                            </span>
                          </div>
                          <div
                            class="ant-transfer-list-body ant-transfer-list-body-with-search"
                          >
                            <div
                              class="ant-transfer-list-body-search-wrapper"
                            >
                              <input
                                class="ant-input ant-transfer-list-search"
                                placeholder="Search here"
                                type="text"
                                value=""
                              />
                              <span
                                class="ant-transfer-list-search-action"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <ul
                              class="ant-transfer-list-content"
                            >
                              <li
                                class="ant-transfer-list-content-item"
                                title="BELFAST TPSIN TEST BROKER 1 - 100118"
                              >
                                <label
                                  class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <span
                                  class="ant-transfer-list-content-item-text"
                                >
                                  BELFAST TPSIN TEST BROKER 1 - 100118
                                </span>
                              </li>
                              <li
                                class="ant-transfer-list-content-item"
                                title="BELFAST TPSIN TEST BROKER 2 - 100119"
                              >
                                <label
                                  class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <span
                                  class="ant-transfer-list-content-item-text"
                                >
                                  BELFAST TPSIN TEST BROKER 2 - 100119
                                </span>
                              </li>
                              <li
                                class="ant-transfer-list-content-item"
                                title="BELFAST TPSIN TEST BROKER 5 - 100122"
                              >
                                <label
                                  class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <span
                                  class="ant-transfer-list-content-item-text"
                                >
                                  BELFAST TPSIN TEST BROKER 5 - 100122
                                </span>
                              </li>
                            </ul>
                          </div>
                        </div>
                        <div
                          class="ant-transfer-operation"
                        >
                          <button
                            class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
                            disabled=""
                            type="button"
                          >
                            <span
                              aria-label="right"
                              class="anticon anticon-right"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="right"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                                />
                              </svg>
                            </span>
                          </button>
                          <button
                            class="ant-btn ant-btn-primary ant-btn-sm ant-btn-icon-only"
                            disabled=""
                            type="button"
                          >
                            <span
                              aria-label="left"
                              class="anticon anticon-left"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="left"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                        <div
                          class="ant-transfer-list"
                          style="width: 380px;"
                        >
                          <div
                            class="ant-transfer-list-header"
                          >
                            <label
                              class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                            >
                              <span
                                class="ant-checkbox"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                            <span
                              aria-label="down"
                              class="anticon anticon-down ant-dropdown-trigger ant-transfer-list-header-dropdown"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="down"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                                />
                              </svg>
                            </span>
                            <span
                              class="ant-transfer-list-header-selected"
                            >
                              2
                               
                              items
                            </span>
                            <span
                              class="ant-transfer-list-header-title"
                            >
                              DMSWeb Broker List
                            </span>
                          </div>
                          <div
                            class="ant-transfer-list-body ant-transfer-list-body-with-search"
                          >
                            <div
                              class="ant-transfer-list-body-search-wrapper"
                            >
                              <input
                                class="ant-input ant-transfer-list-search"
                                placeholder="Search here"
                                type="text"
                                value=""
                              />
                              <span
                                class="ant-transfer-list-search-action"
                              >
                                <span
                                  aria-label="search"
                                  class="anticon anticon-search"
                                  role="img"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="search"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                            <ul
                              class="ant-transfer-list-content"
                            >
                              <li
                                class="ant-transfer-list-content-item"
                                title="BELFAST TPSIN TEST BROKER 3 - 100120"
                              >
                                <label
                                  class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <span
                                  class="ant-transfer-list-content-item-text"
                                >
                                  BELFAST TPSIN TEST BROKER 3 - 100120
                                </span>
                              </li>
                              <li
                                class="ant-transfer-list-content-item"
                                title="BELFAST TPSIN TEST BROKER 4 - 100121"
                              >
                                <label
                                  class="ant-checkbox-wrapper ant-transfer-list-checkbox"
                                >
                                  <span
                                    class="ant-checkbox"
                                  >
                                    <input
                                      class="ant-checkbox-input"
                                      type="checkbox"
                                      value=""
                                    />
                                    <span
                                      class="ant-checkbox-inner"
                                    />
                                  </span>
                                </label>
                                <span
                                  class="ant-transfer-list-content-item-text"
                                >
                                  BELFAST TPSIN TEST BROKER 4 - 100121
                                </span>
                              </li>
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-row ant-row-end action-buttons"
                type="flex"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    <button
                      class="ant-btn"
                      data-testid="bocBtnReset"
                      id="cancel"
                      type="button"
                    >
                      <span>
                        Cancel
                      </span>
                    </button>
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    <button
                      class="ant-btn ant-btn-primary"
                      data-testid="bocBtnApply"
                      id="submit"
                      type="button"
                    >
                      <span>
                        Apply
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </form>
          </div>
          <div
            aria-hidden="true"
            aria-labelledby="rc-tabs-test-tab-Currencies"
            class="ant-tabs-tabpane"
            id="rc-tabs-test-panel-Currencies"
            role="tabpanel"
            style="display: none;"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
